### ISSUE TITLE
Cross Entropy: Check failed: X.ndim() == 2 (4 vs. 2) 

### ISSUE THREAD

**Issue by shankarapailoor** [facebookarchive/caffe2#831]
Hi,

When I try to train my model on the GPU I get this error:

F0619 10:51:07.796715 26017 cross_entropy_op.cu:34] Check failed: X.ndim() == 2 (4 vs. 2) 

I am defining my cross entropy operator as follows:

```python
def AddTrainingOperators(model, softmax, label, device_opts):
        with core.DeviceScope(device_opts):
                xent = model.LabelCrossEntropy([softmax, label], 'xent')
```
The shape of the softmax is (64, 101, 1, 1) and in the cpu implementation I know it checks the first two dimensions of the tensor. How should I go about fixing this?

**Comment by prithayan** [comment]
Hi @shankarapailoor,
I am also facing the same issue. What was the cause for your error ? 
In my experiment, resnet50 model passes with no error, but the model I created fails with the above error.

Thanks!

**Comment by shankarapailoor** [comment]
Hi @prithayan 

The problem is if you have an array of dimension (X, Y, 1, 1) the GPU implementation expects (X, Y) and the CPU ignores the last two dimensions. So to fix this I had to include a squeeze operator before the LabelCrossEntropy.

```python
softmax = model.Squeeze(softmax, somenewnameforsoftmax, dims=[2,3])
xent = model.LabelCrossEntropy([softmax, label], 'xent')
```

This makes the tensor have shape (X, Y)

**Comment by prithayan** [comment]
great, thanks! 
I was actually removing the check, to get the model working, because without the check it just ignores the two dimensions. Thanks for your help!
